packages/ckeditor5-list/src/list/utils.ts

```diff
@@ -8,43 +8,65 @@
  */
 
 import type { Editor } from 'ckeditor5/src/core.js';
-import { ButtonView, type ButtonExecuteEvent } from 'ckeditor5/src/ui.js';
+import { ButtonView, MenuBarMenuListItemButtonView, type ButtonExecuteEvent } from 'ckeditor5/src/ui.js';
 
 /**
- * Helper method for creating a UI button and linking it with an appropriate command.
+ * Helper method for creating toolbar and menu buttons and linking them with an appropriate command.
  *
  * @internal
  * @param editor The editor instance to which the UI component will be added.
  * @param commandName The name of the command.
  * @param label The button label.
  * @param icon The source of the icon.
  */
-export function createUIComponent(
+export function createUIComponents(
 	editor: Editor,
 	commandName: 'bulletedList' | 'numberedList' | 'todoList',
 	label: string,
 	icon: string
 ): void {
-	editor.ui.componentFactory.add( commandName, locale => {
-		const command = editor.commands.get( commandName )!;
-		const buttonView = new ButtonView( locale );
+	editor.ui.componentFactory.add( commandName, () => {
+		const buttonView = _createButton( ButtonView, editor, commandName, label, icon );
 
 		buttonView.set( {
-			label,
-			icon,
 			tooltip: true,
 			isToggleable: true
 		} );
 
-		// Bind button model to command.
-		buttonView.bind( 'isOn', 'isEnabled' ).to( command, 'value', 'isEnabled' );
+		return buttonView;
+	} );
 
-		// Execute command.
-		buttonView.on<ButtonExecuteEvent>( 'execute', () => {
-			editor.execute( commandName );
-			editor.editing.view.focus();
-		} );
+	editor.ui.componentFactory.add( `menuBar:${ commandName }`, () =>
```

Comment:
Or ... is it overwritten on purpose? If so, let's add comment in `listpropertiesui.ts` because it is confusing.

Oh, I see there is a comment there - I didn't notice it because it used to work like that already. I read the comment and I dislike it. It explains it backwards. I'd rewrite this comment, so it is more clear. 

E.g.: "Below we overwrite components registered in `ListUI` plugin. This way, depending on editor configuration, toolbar and menubar will either use the simple component from `ListUI` plugin, or extended component that manges list properties, defined here."